packages/@storefront/flux-capacitor/src/core/reducers/data/navigations.ts

```diff
@@ -65,21 +67,30 @@ export const receiveNavigations = (state: State, navigations: Store.Navigation[]
 };
 
 // tslint:disable-next-line max-line-length
-export const selectRefinement = (state: State, { navigationId, index: refinementIndex }: Actions.Payload.Navigation.Refinement) => {
-  if (navigationId && refinementIndex != null) {
-    return {
-      ...state,
-      byId: {
-        ...state.byId,
-        [navigationId]: {
-          ...state.byId[navigationId],
-          selected: state.byId[navigationId].selected.concat(refinementIndex),
-        },
+export const selectMultipleRefinements = (state: State, { navigationId, indices }: Actions.Payload.Navigation.MultipleRefinements) => {
+  if (!navigationId) {
+    return state;
+  }
+
+  return {
+    ...state,
+    byId: {
+      ...state.byId,
+      [navigationId]: {
+        ...state.byId[navigationId],
+        selected: Array.from(new Set(state.byId[navigationId].selected.concat(indices))),
       },
-    };
-  } else {
+    },
+  };
+};
+
+// tslint:disable-next-line max-line-length
+export const selectRefinement = (state: State, { navigationId, index: refinementIndex }: Actions.Payload.Navigation.Refinement) => {
+  if (!navigationId || refinementIndex == null) {
```

Comment:
I don't think you need to check this because 1) it must be passed in as a parameter to the action creator and 2) the action creator goes through a validator that checks whether it exists. If you want to check whether the index is a number, that should be done in the action validator

Reply:
Agreed, moved the checks into the validators. Resolving conversation.